jaeger_client/span.py

```diff
@@ -67,11 +71,15 @@ def finish(self, finish_time=None):
         :param finish_time: an explicit Span finish timestamp as a unix
             timestamp per time.time()
         """
-        if not self.is_sampled():
-            return
+        with self.update_lock:
```

Comment:
I suggest keeping the is_sampled check above. Otherwise you're introducing a mutex lock for all spans, even unsampled ones, which is an unnecessary performance hit.

Reply:
Good point, moved it above. So i suppose we're ok with not marking span as finished if its not sampled firstly.